pkg/resourceprovider/provider.go

```diff
@@ -159,7 +160,14 @@ func (p *resourceProvider) GetPodMetrics(pods ...*metav1.PartialObjectMetadata)
 	resultsByNs := make(map[string][]nsQueryResults, len(podsByNsBatched))
 	for result := range resChan {
 		if result.err != nil {
-			klog.Errorf("unable to fetch metrics for pods in namespace %q, skipping: %v", result.namespace, result.err)
+			if promErr, ok := result.err.(*client.Error); ok {
+				klog.Errorf("unable to fetch metrics for pods in namespace %q, error: %s", result.namespace, promErr.Error())
```

Comment:
ditto: Log all the info we have here, i.e. status code, query, error message, type